### PR TITLE
[FIX] Convert datetime into date with timezone

### PR DESCRIPTION
**Pull request by ddufresne**
Before this fix, as a user, if you entered a timesheet with a given datetime,
the timesheet could actually be stored as another date in the database.

This falsifies the grid view and other reports that group analytic lines per date.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I don't really understand the problem...

@Yajo do you?

**Review by foutoucour — APPROVED** [review]

File: project_timesheet_time_control/models/account_analytic_line.py
Comment (yajo) [inline project_timesheet_time_control/models/account_analytic_line.py:35]: This should be enough, no need to add an extra method:

```suggestion
            return dict(
                vals,
                date=fields.Date.to_date(fields.Datetime.to_datetime(vals['date_time']))
            )
        return vals
```

File: project_timesheet_time_control/tests/test_project_timesheet_time_control.py
Comment (yajo) [inline project_timesheet_time_control/tests/test_project_timesheet_time_control.py:74]: @pedrobaeza Notice this test.

If an AAL is created with this timezone and this time, the AAL date field should be mirrored as being the previous day, because DB dates are always stored in UTC, and technically, in UTC it would be day 23.

**Review by yajo — CHANGES_REQUESTED** [review]

File: project_timesheet_time_control/models/account_analytic_line.py
Comment (ddufresne) [reply project_timesheet_time_control/models/account_analytic_line.py:35]: @Yajo I disagree about your suggestion. First, it breaks the behavior. Second, the extra method makes the intension more explicit.

Comment (yajo) [reply project_timesheet_time_control/models/account_analytic_line.py:35]: What does it break?

Comment (ddufresne) [reply project_timesheet_time_control/models/account_analytic_line.py:35]: @Yajo it does not use the timezone of the user. Your solution is similar to the original behavior before this pull request.

Comment (yajo) [reply project_timesheet_time_control/models/account_analytic_line.py:35]: OK, how about this?

```suggestion
            ts = fields.Datetime.to_datetime(vals['date_time'])
            ts = fields.Date.context_today(ts)
            return dict(vals, date=ts)
```

Comment (ddufresne) [reply project_timesheet_time_control/models/account_analytic_line.py:35]: @Yajo ok, but there's a small error in your proposal:

ts = fields.Date.context_today(ts)

should be

ts = fields.Date.context_today(self, ts)

**Comment by dreispt** [comment]
@ddufresne We should use the builtin methods of the Date / Datetime fields, so @Yajo is on the right track here.

**Comment by ddufresne** [comment]
@dreispt Yajo's solution does not make the unit tests pass.

Otherwise, you are implying that my solution does not use the build-in methods, which is not the case.

**Comment by dreispt** [comment]
@ddufresne Yes you are, and your code is passing the tests checking the timezone.
So this seems correct to me.

**Review by dreispt — APPROVED** [review]

**Comment by ddufresne** [comment]
@dreispt thanks!

**Review by yajo — APPROVED** [review]
Yes, I didn't mean your code was wrong. I just think that adding an extra method to fix this is not worth it. But nevermind, do as you please. :man_shrugging:

**Comment by pedrobaeza** [comment]
Oh, wait, please squash the 2 commits.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Squash commits

**Review by marcelsavegnago — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash commits for merging this.

**Comment by pedrobaeza** [comment]
Why have you merged this?

**Comment by ddufresne** [comment]
@pedrobaeza what do you mean?

You told to squash before merging. I squashed before merging.

The PR is approved by 4 people other than me.

**Comment by pedrobaeza** [comment]
You must squash in your branch, but not merge the PR. That's done with the bot, and now there's no version bumping due to this. Please don't do this next time and just squash in your local branch.

**Comment by ddufresne** [comment]
@pedrobaeza my appologies, did not know about this.

Couldn't this robot squash the PR by itself before merging?

**Comment by pedrobaeza** [comment]
No, it's a limitation

**Comment by yajo** [comment]
FTR https://github.com/OCA/oca-github-bot/issues/43

**Pull request merged**
